src/com/goide/inspections/GoExportedOwnDeclarationInspection.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.text.StringUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class GoExportedOwnDeclarationInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitConstDeclaration(@NotNull GoConstDeclaration o) {
+        if (!(o.getParent() instanceof GoFile)) {
+          return;
+        }
+
+        for (int i = 0; i < o.getConstSpecList().size(); i++) {
+          int index = 0;
```

Comment:
So where is the test for 

```go
// valid code

const (
    Foo = 1
    Bar = 1
)

// warning code

const (
    Foo = 1
    Bar, Baz = 1 // Baz should be declared in it's own block
)
```

I didn't check, but I think your implementation fails on this code